### PR TITLE
Add API forwarding support for HyperV

### PR DESCRIPTION
**Pull request by ashley-cui**
Provides Docker API client access, allowing compose to work by default for HyperV. Basically the HyperV equiv of the work done here by #12916.

[NO NEW TESTS NEEDED]

Closes https://github.com/containers/podman/issues/20761

```release-note
Fixed a bug where Podman machine on HyperV did not allow API forwarding
```

### CODE REVIEW

**Comment by ashley-cui** [comment]
@n1hility @jeffmaury PTAL. Tested and it seems to work for me.

**Comment by martinpitt** [comment]
Yup, that's a regression in criu in podman-next. We see it in our [nightly runs against podman-next](https://github.com/cockpit-project/cockpit-podman/issues/1495) as well, it [fails on the same "criu restore" check](https://cockpit-logs.us-east-1.linodeobjects.com/pull-0-20231202-015044-39de27b4-fedora-39-podman-next/log.html). I'll investigate/report on Monday.

**Comment by rhatdan** [comment]
LGTM

File: pkg/machine/hyperv/machine.go [hunk 1]
```diff
@@ -610,6 +613,35 @@ func (m *HyperVMachine) Start(name string, opts machine.StartOptions) error {
...
+				fmt.Printf("\nAnother process was listening on the default Docker API pipe address.\n")
+				fmt.Printf("You can still connect Docker API clients by setting DOCKER HOST using the\n")
+				fmt.Printf("following powershell command in your terminal session:\n")
+				fmt.Printf("\n\t$Env:DOCKER_HOST = '%s'\n", pipeName)
+				fmt.Printf("\nOr in a classic CMD prompt:\n")
+				fmt.Printf("\n\tset DOCKER_HOST=%s\n", pipeName)
+				fmt.Printf("\nAlternatively, terminate the other process and restart podman machine.\n")
+			}
+		}
+	}
```
Comment (Luap99) [inline pkg/machine/hyperv/machine.go:644]: Is the text duplicated? If so please deduplicate.

**Comment by jeffmaury** [comment]
LGTM but I add to add ssh-keygen to get the machine created

**Comment by n1hility** [comment]
Sorry this one fell through the cracks for me and the very late review. This LGTM 

This is not a blocker at all, but just mentioning to factor in / consider for the future. Since the hyperv code uses gvproxy for ssh routing  / port forwarding, if we wanted to save a process and drop usage of win-sshproxy the ssh options on gvproxy could be used for this purpose instead (like what we do on mac). It's still a trade-off to some extent since it would mean differences between the hyperv and wsl code.

**Comment by jeffmaury** [comment]
I have 2 remarks:

- when the machine is stopped the win-sshproxy processed is not stopped
- If I run podman compose against https://github.com/docker/awesome-compose/tree/master/flask-redis, I have the following error (which I don't have with a WSL machine):
```
C:\Users\Jeff\work\src\github.com\docker\awesome-compose\flask-redis>podman compose up -d
>>>> Executing external compose provider "C:\\Users\\Jeff\\AppData\\Local\\Microsoft\\WindowsApps\\docker-compose.exe". Please refer to the documentation for details. <<<<

[+] Running 2/0
 ✔ Network flask-redis_default    Created                                                                          0.0s
 ✔ Container flask-redis-redis-1  Created                                                                          0.1s
 - Container flask-redis-web-1    Creating                                                                         0.0s
Error response from daemon: fill out specgen: C:\Users\Jeff\work\src\github.com\docker\awesome-compose\flask-redis:/code:rw: incorrect volume format, should be [host-dir:]ctr-dir[:option]
Error: executing C:\Users\Jeff\AppData\Local\Microsoft\WindowsApps\docker-compose.exe up -d: exit status 1
```

**Comment by rhatdan** [comment]
@ashley-cui needs a rebase

**Comment by ashley-cui** [comment]
@jeffmaury 
> * when the machine is stopped the win-sshproxy processed is not stopped

Updated, thanks!
> * If I run podman compose against https://github.com/docker/awesome-compose/tree/master/flask-redis, I have the following error (which I don't have with a WSL machine):

This is another followup issue, where the parsing of mounts on HyperV requires absolute paths. Will address this later.

**Comment by ashley-cui** [comment]
Should be good to go, I think the mac failure is unrelated.

**Pull request merged**